### PR TITLE
Correct documentation for Date.match()

### PR DESCRIPTION
**Pull request by CameronD73**
This is just documentation changes for date.py.
1. Correct the description of how match() works.
2. Extend detail of various options for instantiation.
3. Clarify operation of set_yr_mon_day_offset()  and copy_offset_ymd()  on compound dates.

Fixes bug [#13428](https://gramps-project.org/bugs/view.php?id=13428).

### CODE REVIEW

**Comment by QuLogic** [comment]
I think you may have accidentally included your changes for #1772 here.

**Comment by CameronD73** [comment]
> I think you may have accidentally included your changes for #1772 here.
Yes, I thought that might have happened - basically I don't know what I am doing. I have used git for trivial things but nothing requiring PRs.

I merged the 1772 branch into the local maint/gramps52 then created the new branch off that.
If you want to reject this I can have another go.   It is not obvious to me how I can withdraw it.

**Comment by QuLogic** [comment]
No need for that; you can rebase and drop the extra commits from here since they are separate branches:
```
$ git switch Bug0013428
$ git rebase --interactive maintenance/gramps52
```
In the editor that opens, delete every line except for the last one about bug 13428, save and exit to run the rebase.


After the rebase, you can check the log should show only the one commit on this branch:
```
$ git log maintenance/gramps52..Bug0013428
```
And if everything is okay, force push to GitHub:
```
$ git push --force-with-lease origin Bug0013428
```
If not okay, then reset back to the copy you have here:
```
$ git reset --hard origin/Bug0013428
```

**Comment by CameronD73** [comment]
Thanks...  OK, I think I have managed to fix it - or at least I can say it has changed.

Your rebase command resulted in a file with only the single commit, so I tried
`git rebase --interactive 6442fa3d  maintenance/gramps52`
This gave me a few other commits to remove, and set the bug branch off the previous point in maint/gramps52, as I was hoping for.
But the bug branch itself still had the merges that had been removed from the maint branch.
After another 
` git rebase --interactive 6442fa3d  Bug001342`
I seem to have achieved the state I was after, following the force-push operations.

**Comment by QuLogic** [comment]
> I merged the 1772 branch into the local maint/gramps52 then created the new branch off that.

Sorry, I missed this line; you probably don't want to merge your branches into upstream branches. Just wait until they are merged in this repo.

To be more explicit, the first step I gave could have been `git rebase --interactive upstream/maintenance/gramps52` (assuming you have this repo defined as a remote named `upstream`), and then it should have worked based on differences from this repo. But everything is good now, so there's no need to fiddle with it.

**Review by dsblank — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**
